controllers/core/event_controller.go

```diff
@@ -47,7 +49,20 @@ const (
 	ReconcileEventsLatencyMetrics = "reconcile_events"
 	LabelOperation                = "operation"
 	LabelTime                     = "time"
+	LabelCacheHit                 = "cache_hit"
+	LabelCacheMiss                = "cache_miss"
+	LabelCacheSetErr              = "cache_set_error"
+	CacheHitMetricsValue          = "NODE_CACHED"
+	CacheMissMetricsValue         = "NODE_NOT_CACHED"
+	CacheErrMetricsValue          = "NODE_CACHE_ERROR"
 	EventRegardingKind            = "Node"
+	SGPPrefix                     = "SGP"
+	CNPrefix                      = "CN"
+
+	// customized configurations for BigCache
+	InstancesCacheTTL     = 30 * time.Minute // scaling < 1k nodes should be under 20 minutes
+	InstancesCacheShards  = 32               // must be power of 2
```

Comment:
Looks like the key is `(SGPPrefix | CNPrefix) + hostId)`. For 1k nodes and 5 security groups, the space < 2^13, so I think 32 shards may be too many. 16 is probably sufficient and more efficient in the average case.

Reply:
Yeah, I was thinking the same thing. Sharding 32 might bring too much overhead. I would wait for other members' suggestions as well.